### PR TITLE
Cleanup and more tests

### PR DESCRIPTION
**Pull request by eatonphil**
This is a messy diff but the biggest thing I changed @Posrabi was to make `-s` a string flag not a boolean flag and also to always read from stdin immediately so there are not special cases when calculating hashes or importing files.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -471,36 +420,51 @@ func _main() error {
...
 		}
 	}
 
 	ec := runner.NewEvalContext(*runner.DefaultSettings, tmpDir)
-	if !cacheSettings.CachePresent || !cacheSettings.Enabled {
-		err := importFiles(cacheSettings.Enabled, files, &ec, project)
-		if err != nil {
-			return err
+	// When dumping schema, need to injest even if cache is on.
+	if !args.cacheSettings.CachePresent || !args.cacheSettings.Enabled || lastNonFlagArg == "" {
```
Comment (Posrabi) [inline main.go:434]: I think we might have to also truncate the cache file if we want to reinjest.

Comment (eatonphil) [reply main.go:434]: Is that something I accidentally got rid of or you are just noticing it now?

Comment (Posrabi) [reply main.go:434]: I just noticed it now. The old code didn't reingest so I didn't include that.

Comment (eatonphil) [reply main.go:434]: The old code had to have reingested, I didn't change that.

If the cache was invalid it reingested

Comment (Posrabi) [reply main.go:433]: Nah, it just checked that if the cache is present or not.

**Comment by eatonphil** [comment]
Two things that I noticed @Posrabi  was passing in the files array to modify it and passing project to append panels to it.

That is a good idea in performance sensitive code but by default you should try not to modify pointers within a function and try to just pass back objects and have the caller append it to an array or something.

Just easier to read, easier to test and less likely to be buggy that way.

**Comment by Posrabi** [comment]
I see, thanks for the feedback!

**Pull request merged**